library/core/src/clone.rs

```diff
@@ -86,6 +86,24 @@
 /// }
 /// ```
 ///
+/// If you `derive`:
+///
+/// ```
+/// #[derive(Clone)]
+/// struct Generate<T>(fn() -> T);
+/// ```
+///
+/// the auto-derived implementation will be **invalid**, because it will become:
```

Comment:
I don't think "invalid" is a good thing to say here, since that code compiles fine <https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=63f01a475d60aa4896f193a3a887044b>.

That might not be what you *want*, but it's not *invalid*, so I think this section should better elaborate what it's trying to say.

@rustbot author

Reply:
Thank you for the feedback! I have now pushed a more fleshed out explanation. Looking forward to your new feedback. Here is how the new version looks when rendered:

![Skärmbild 2023-07-30 195929](https://github.com/rust-lang/rust/assets/115040/6b61e1de-0555-46da-9790-3e2d940f76ce)

@rustbot ready